File: solutions/c/luhn/1/luhn.c
#include "luhn.h"
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>
bool luhn(const char *num) {
  bool flag = false;
  int length = strlen(num);
  int *new_arr = malloc(length * sizeof(int));
  int count = 0;
  for (int i = length - 1; i > -1; i--) {
    if (num[i] == ' ') {
      continue;
    }
    if (num[i] - '0' >= 10 || num[i] - '0' < 0) {
      return false;
    }
    if (flag) {
      new_arr[i] = (num[i] - '0') * 2;
      if (new_arr[i] > 9) {
        new_arr[i] = new_arr[i] - 9;
      }
    } else {
      new_arr[i] = num[i] - '0';
    }
    flag = !flag;
    count++;
  }
  if (count <= 1) {
    return false;
  }
  int sum = 0;
  for (int i = 0; i < length; i++) {
    sum = sum + new_arr[i];
  }
  if (sum % 10 == 0) {
    return true;
  }
  return false;
}

File: solutions/python/luhn/1/luhn.py
class Luhn:
    def __init__(self, card_num):
        self.flag = False
        self.card_num = card_num
        pass

    def valid(self):
        card_num = self.card_num.replace(" ","")
        flag = self.flag
        arr_num = list(card_num)
        length = len(arr_num)
        if length <=1:
            return False
        for i in range(length-1,-1,-1):
            if not arr_num[i].isdigit():
                return False
            if flag:
                arr_num[i] = int(arr_num[i]) * 2
                if arr_num[i] > 9:
                    arr_num[i] = arr_num[i]-9
            else:
                arr_num[i] = int(arr_num[i])
            flag = not flag
        add = sum(arr_num)
        if not (add % 10):
            return True
        return False
            
